File: server/src/main/java/cn/iinti/atom/system/SystemConfiguration.java
package cn.iinti.atom.system;

import cn.iinti.atom.BuildConfig;
import com.baomidou.mybatisplus.extension.plugins.inner.PaginationInnerInterceptor;
import io.swagger.v3.oas.models.ExternalDocumentation;
import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.info.Info;
import io.swagger.v3.oas.models.info.License;
import io.swagger.v3.oas.models.parameters.Parameter;
import io.swagger.v3.oas.models.parameters.QueryParameter;
import org.apache.commons.lang3.StringUtils;
import org.springdoc.core.customizers.OperationCustomizer;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.ArrayList;
import java.util.List;

@Configuration
public class SystemConfiguration {
    @Bean
    public PaginationInnerInterceptor paginationInterceptor() {
        return new PaginationInnerInterceptor();
    }

    @Bean
    public OpenAPI springShopOpenAPI() {
        return new OpenAPI()
                .info(new Info().title("Atom")
                        .description("atom系统")
                        .version("v" + BuildConfig.versionName)
                        .license(new License().name("Apache 2.0")
                                .url("https://atom.iinti.cn/atom-doc")))
                .externalDocs(new ExternalDocumentation()
                        .description("iinti")
                        .url("https://iinti.cn"));
    }

    @Bean
    public OperationCustomizer apiCustomizer() {
        return (operation, handlerMethod) -> {
            LoginRequired loginRequired = handlerMethod.getMethod().getAnnotation(LoginRequired.class);
            if (loginRequired == null) {
                return operation;
            }
            List<String> newSummary = new ArrayList<>();
            if (loginRequired.forAdmin()) {
                newSummary.add("AdminOnly");
            }
            if (loginRequired.apiToken()) {
                newSummary.add("SupportApiToken");
            }
            String summary = operation.getSummary();
            if (StringUtils.isNotBlank(summary)) {
                newSummary.add(summary);
            }
            operation.setSummary(StringUtils.join(newSummary, " "));
            Parameter parameter = new QueryParameter()
                    .name(BuildConfig.userLoginTokenKey)
                    .description("接口Token")
                    .required(true)
                    .allowEmptyValue(false);
            operation.addParametersItem(parameter);

            return operation;
        };
    }
}

File: server/src/main/kotlin/cn/iinti/atom/system/SystemConfiguration.kt
package cn.iinti.atom.system

import cn.iinti.atom.BuildInfo
import com.baomidou.mybatisplus.extension.plugins.inner.PaginationInnerInterceptor
import io.swagger.v3.oas.models.ExternalDocumentation
import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Info
import io.swagger.v3.oas.models.info.License
import io.swagger.v3.oas.models.parameters.QueryParameter
import org.apache.commons.lang3.StringUtils
import org.springdoc.core.customizers.OperationCustomizer
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class SystemConfiguration {
    @Bean
    fun paginationInterceptor(): PaginationInnerInterceptor {
        return PaginationInnerInterceptor()
    }

    @Bean
    fun springShopOpenAPI(): OpenAPI {
        return OpenAPI()
            .info(
                Info()
                    .title("Atom")
                    .description("atom系统")
                    .version("v${BuildInfo.versionName}")
                    .license(
                        License()
                            .name("Apache 2.0")
                            .url("https://atom.iinti.cn/atom-doc")
                    )
            )
            .externalDocs(
                ExternalDocumentation()
                    .description("iinti")
                    .url("https://iinti.cn")
            )
    }

    @Bean
    fun apiCustomizer(): OperationCustomizer {
        return OperationCustomizer { operation, handlerMethod ->
            val loginRequired = handlerMethod.method.getAnnotation(LoginRequired::class.java)
                ?: return@OperationCustomizer operation
            val newSummary = ArrayList<String>()
            if (loginRequired.forAdmin) {
                newSummary.add("AdminOnly")
            }
            if (loginRequired.apiToken) {
                newSummary.add("SupportApiToken")
            }
            val summary = operation.summary
            if (StringUtils.isNotBlank(summary)) {
                newSummary.add(summary)
            }
            operation.summary = StringUtils.join(newSummary, " ")
            val parameter = QueryParameter()
                .name(BuildInfo.userLoginTokenKey)
                .description("接口Token")
                .required(true)
                .allowEmptyValue(false)
            operation.addParametersItem(parameter)

            operation
        }
    }
}